### ISSUE TITLE
Optimizer causes ambiguous identifier error on `for*/fold`

### ISSUE THREAD

**Issue by florence** [racket/typed-racket#576]
### What version of Racket are you using?
`6.9.0.6--2017-2-22`

### What program did you run?
```
#lang typed/racket
(for*/fold ([mx 0.0+0.0i])
          ([_ (in-range 0 1)])
  (+ (magnitude mx) 0.0+0.0i))
```

### What should have happened?
returned `0.0+0.0i`. This happens if ` #:no-optimize` is given on the lang line

### If you got an error message, please include it here.
```
 mx: identifier's binding is ambiguous
  context...:
  matching binding...:
  matching binding...:
  matching binding...:
```

**Comment by florence** [comment]
The commit that closes this seems to just be tests? (An after updating I still get the error).

**Comment by stamourv** [comment]
I pushed your commit along as well. I don't see the problem anymore. Did you re-run `raco setup`?

**Comment by florence** [comment]
Yes. My commit fixed a different problem. This one is an ambiguous binding
error from for*/fold+magnitude, not magnitude returning the wrong value on
some numbers. I'm not sure how to fix this one.

**Comment by stamourv** [comment]
Oh, I closed the wrong thing. My bad.
